Tasks/ServiceFabricDeploy/Tests/CreateDiffPkg.ps1

```diff
@@ -0,0 +1,101 @@
+Register-Mock Get-VstsInput { "false" } -- -Name unregisterUnusedVersions
+Register-Mock Get-VstsInput { "false" } -- -Name configureDockerSettings
+Register-Mock Get-VstsInput { $true }  --  -Name useDiffPackage
+Register-Mock Get-VstsInput { $diffPkgAssetsPath } -- -Name diffPackagePath -Require
+
+# Setup file resolution
+Register-Mock Find-VstsFiles { $publishProfilePath } -- -LegacyPattern $publishProfilePath
+Register-Mock Find-VstsFiles { $applicationPackagePath } -- -LegacyPattern $applicationPackagePath -IncludeDirectories
+
+Register-Mock Assert-VstsPath
+Register-Mock Test-Path { $true } -- "HKLM:\SOFTWARE\Microsoft\Service Fabric SDK"
+
+# Setup mock VSTS service endpoint
+$vstsEndpoint = @{
+    "Auth" = @{
+        "Scheme" = "None"
+    }
+}
+Register-Mock Get-VstsEndpoint { $vstsEndpoint } -- -Name $serviceConnectionName -Require
+
+# Setup mock for connection to cluster
+Register-Mock Connect-ServiceFabricCluster { $null } -- -ConnectionEndpoint "test"
+
+# Setup mock registry settings
+$regKeyObj = @{
+    "FabricSDKPSModulePath" = $serviceFabricSdkModulePath
+}
+Register-Mock Get-ItemProperty { $regKeyObj } -- -Path "HKLM:\SOFTWARE\Microsoft\Service Fabric SDK" -Name FabricSDKPSModulePath
+
+Register-Mock Get-ApplicationNameFromApplicationParameterFile { $appName } -- "$PSScriptRoot\data\ApplicationParameters.xml"
+
+$app = @{
+    "ApplicationTypeName" = $applicationTypeName;
+    "ApplicationTypeVersion" = $applicationTypeVersion
+}
+Register-Mock Get-ServiceFabricApplication { $app } -- -ApplicationName $appName
+$publishArgs = @("-ApplicationParameterFilePath:", "$PSScriptRoot\data\ApplicationParameters.xml", "-OverwriteBehavior:", $overwriteBehavior, "-ApplicationPackagePath:", "aaa", "-ErrorAction:", "Stop", "-Action:", "RegisterAndCreate")
+Register-Mock Publish-NewServiceFabricApplication -Arguments $publishArgs
+
+$serviceType1 = @{
+    "ServiceManifestName" = "Stateless1Pkg"
+}
+$serviceType2 = @{
+    "ServiceManifestName" = "Stateless2Pkg"
+}
+$serviceTypes = @($serviceType1, $serviceType2)
+$serviceManifest = '<ServiceManifest Name="Stateless1Pkg" Version="1.0.0">' +
+  '<ServiceTypes>' +
+    '<StatelessServiceType ServiceTypeName="Stateless1Type" />' +
+  '</ServiceTypes>' +
+  '<CodePackage Name="Code" Version="1.0.0">' +
+  '</CodePackage>' +
+  '<ConfigPackage Name="Config" Version="1.0.0" />' +
+'</ServiceManifest>'
+
+Register-Mock Get-ServiceFabricServiceType {$serviceTypes} -- -ApplicationTypeName $applicationTypeName -ApplicationTypeVersion $applicationTypeVersion
+Register-Mock Get-ServiceFabricServiceManifest {$serviceManifest} -- -ApplicationTypeName $applicationTypeName -ApplicationTypeVersion $applicationTypeVersion -ServiceManifestName "Stateless1Pkg"
+Register-Mock Get-ServiceFabricServiceManifest {$serviceManifest} -- -ApplicationTypeName $applicationTypeName -ApplicationTypeVersion $applicationTypeVersion -ServiceManifestName "Stateless2Pkg"
```

Comment:
The manifest returned is wrong for Stateless2Pkg as it name's itself Stateless1Pkg